src/viam/sdk/services/mlmodel/server.cpp

```diff
@@ -34,184 +35,134 @@ void MLModelServiceServer::register_server(std::shared_ptr<Server> server) {
 ::grpc::Status MLModelServiceServer::Infer(
     ::grpc::ServerContext* context,
     const ::viam::service::mlmodel::v1::InferRequest* request,
-    ::viam::service::mlmodel::v1::InferResponse* response) noexcept try {
-    if (!request) {
-        return {::grpc::StatusCode::INVALID_ARGUMENT, "Called [Infer] without a request"};
-    };
-
-    std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
-
-    if (!rb) {
-        return {::grpc::UNKNOWN, "resource not found: " + request->name()};
-    }
-
-    std::shared_ptr<MLModelService> mlms = std::dynamic_pointer_cast<MLModelService>(rb);
-
-    if (!request->has_input_tensors()) {
-        return {::grpc::StatusCode::INVALID_ARGUMENT, "Called [Infer] with no inputs"};
-    }
-
-    AttributeMap extra;
-    if (request->has_extra()) {
-        extra = struct_to_map(request->extra());
-    }
-
-    const auto md = mlms->metadata(extra);
-    MLModelService::named_tensor_views inputs;
-    for (const auto& input : md.inputs) {
-        const auto where = request->input_tensors().tensors().find(input.name);
-        if (where == request->input_tensors().tensors().end()) {
-            // Ignore any inputs for which we don't have metadata, since
-            // we can't validate the type info.
-            //
-            // TODO: Should this be an error? For now we just don't decode
-            // those tensors.
-            continue;
+    ::viam::service::mlmodel::v1::InferResponse* response) noexcept {
+    return make_service_helper<MLModelService>(
+        "MLModelServiceServer::Infer", this, request)([&](auto& helper, auto& mlms) {
+        if (!request->has_input_tensors()) {
+            return helper.fail(::grpc::INVALID_ARGUMENT, "Called with no input tensors");
         }
-        auto tensor = mlmodel_details::make_sdk_tensor_from_api_tensor(where->second);
-        const auto tensor_type = MLModelService::tensor_info::tensor_views_to_data_type(tensor);
-        if (tensor_type != input.data_type) {
-            std::ostringstream message;
-            using ut = std::underlying_type<MLModelService::tensor_info::data_types>::type;
-            message << "Tensor input `" << input.name << "` was the wrong type; expected type "
-                    << static_cast<ut>(input.data_type) << " but got type "
-                    << static_cast<ut>(tensor_type);
-            return ::grpc::Status(grpc::INVALID_ARGUMENT, message.str());
+
+        const auto md = mlms->metadata({});
```

Comment:
Previously this passed `extra` into this call to `metadata` but I think that was unnecessary. The `extra` parameters here are intended for the `Infer` call. The implementation of `Infer` needs access to the metadata, but that doesn't mean it should also be given the same `extra` params that were passed to `Infer`.

Reply:
The only actual implementations of that method I've seen are in the tflite module and the mock mlmodel service, neither of which use `extra`. I don't really see why it's a necessary argument to `metadata` either, so I'm fine removing it unless @stuqdog disagrees.

Reply:
Sounds reasonable to me! I added `extra` there out of an abundance of caution when adding `extra` overloads, not out of some principled stance on it being necessary to `metadata`.